File: testbed_ruby/test/attributes/constraint_test.rb

require 'test/unit'
require './src-gen-umple/boat_a'
require './src-gen-umple/boat_b'

module CruiseAttributesTest
  class ConstraintTest < Test::Unit::TestCase

    def test_checkConstraint
      boat = BoatA.new(20)
      assert_equal(false, boat.set_age(18))
      assert_equal(true, boat.set_age(19))
    end

    def test_checkNegation
      boat = BoatB.new(2)
      assert_equal(true, boat.set_age(18))
      assert_equal(false, boat.set_age(19))
    end


  end

end

File: testbed/test/cruise/attributes/test/ConstraintTest.java
package cruise.attributes.test;

import org.junit.*;

public class ConstraintTest
{

  @Test
  public void checkConstraint()
  {
    BoatA boat = new BoatA(20);
    Assert.assertEquals(false, boat.setAge(18));
    Assert.assertEquals(true, boat.setAge(19));
  }

  @Test
  public void checkNegation()
  {
    BoatB boat = new BoatB(2);
    Assert.assertEquals(true, boat.setAge(18));
    Assert.assertEquals(false, boat.setAge(19));
  }

}
